src/Diagnostics.DataProviders/MdmClient/MdmClient.cs

```diff
@@ -353,6 +353,7 @@ private static HttpClient CreateHttpClient(string thumbprint)
                         (e.ResponseStatusCode != null && e.ResponseStatusCode != HttpStatusCode.ServiceUnavailable) ||
                         i == numAttempts)
                     {
+                        DiagnosticsETWProvider.Instance.LogDataProviderMessage(RequestId, "MdmClient", $"GetResponse Failure. Latency: {stopWatch.Elapsed}; Number of attempts: {i}; Exception: {e.ToString()}");
```

Comment:
If you log it here then we won't be able to track _all_ the retries from MDM. Consider logging this within a finally clause.